### DISCUSSION TITLE
Which DML model to use for a binary treatment and a binary outcome?

### DISCUSSION THREAD

**Discussion by CJames1261** [DoubleML/doubleml-for-py#234] [Q&A]
Hi I am trying to perform causal analysis with an outcome and treatment that are both binary. Here is my code:

'''
obj_dml_data = dml.DoubleMLData(data,y_col='binary_outcome',d_cols='binary_treatment') 

dml_mlplr  = dml.DoubleMLPLR(obj_dml_data,
                                 ml_l = RandomForestClassifier(),
                                 ml_m = RandomForestClassifier(),
                                 n_folds = 5)

summary = dml_mlplr.fit(store_predictions=True).summary

summary
"""

Outcome:
data = {
    'oef': ['binary_treatment'],
    'std err': [0.509206],
    't': [44.761216],
    'P>|t|': [0.0],
    '2.5 %': [0.48691],
    '97.5 %': [0.531503]
}

I do get an output but I see this warning and would like to get some clarification on what it means:
'Learner provided for ml_l is probably invalid: RandomForestClassifier() is (probably) no regressor.'

What exactly does this mean? do I need to you a different dml model other than DoubleMLPLR? 
Thanks!

**Comment by SvenKlaassen** [comment]
Currently, we assume that the learner for ml_l ist a regressor see [DoubleMLPLR](https://docs.doubleml.org/dev/api/generated/doubleml.DoubleMLPLR.html) as the `predict()` method is used to obtain predictions for $\mathbb{E}[Y|X]$. This can lead to some misleading results if a classifier is used (here we would have to rely on `predict_proba()`.

You should use a regressor or [external predictions](https://docs.doubleml.org/dev/guide/learners.html#advanced-external-predictions) for `ml_l`, but I think it would be a good idea to extend the model to correctly use classifiers.

**Reply by CJames1261** [reply]
That would be a great feature to add, If you guys implement it then please let me know! Thanks
